File: app/src/main/java/com/iitk/proxyLogin/AlarmReceiver.java
package com.iitk.proxyLogin;

import android.content.Context;
import android.content.Intent;
import android.os.SystemClock;
import android.support.v4.content.WakefulBroadcastReceiver;
import android.util.Log;

/**
 * Created by kshivang on 15/04/17.
 *
 */

public class AlarmReceiver extends WakefulBroadcastReceiver {
    private static final String TAG;

    static {
        TAG = AlarmReceiver.class.getSimpleName();
    }

    public void onReceive(Context context, Intent intent) {
        if ("proxy.service.ALARM_BROADCAST".equals(intent.getAction())) {
            Intent proxyServiceIntent = new Intent(context, ProxyService.class);
            proxyServiceIntent.setAction("proxy.service.ALARM_BROADCAST");
            Log.i(TAG, "#### Launching ProxyService for " +
                    intent.getAction() + " @ " + SystemClock.elapsedRealtime());
            WakefulBroadcastReceiver.startWakefulService(context, proxyServiceIntent);
        }
    }
}

File: app/src/main/java/com/iitk/proxyLogin/receiver/AlarmReceiver.kt
package com.iitk.proxyLogin.receiver

import android.content.Context
import android.content.Intent
import android.os.SystemClock
import android.support.v4.content.WakefulBroadcastReceiver
import android.util.Log

import com.iitk.proxyLogin.service.ProxyService

/**
 * Created by kshivang on 15/04/17.

 */

class AlarmReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if ("proxy.service.ALARM_BROADCAST" == intent.action) {
            val proxyServiceIntent = Intent(context, ProxyService::class.java)
            proxyServiceIntent.action = "proxy.service.ALARM_BROADCAST"
            Log.i(TAG, "#### Launching ProxyService for " +
                    intent.action + " @ " + SystemClock.elapsedRealtime())
            WakefulBroadcastReceiver.startWakefulService(context, proxyServiceIntent)
        }
    }

    companion object {
        private val TAG: String = AlarmReceiver::class.java.simpleName

    }
}
